jgiven-core/src/main/java/com/tngtech/jgiven/report/asciidoc/AsciiDocReportGenerator.java

```diff
@@ -1,196 +1,186 @@
 package com.tngtech.jgiven.report.asciidoc;
 
-import java.io.File;
-import java.io.PrintWriter;
-import java.util.List;
-
-import com.google.common.collect.Lists;
 import com.google.common.io.Files;
 import com.tngtech.jgiven.impl.util.PrintWriterUtil;
-import com.tngtech.jgiven.impl.util.ResourceUtil;
-import com.tngtech.jgiven.impl.util.WordUtil;
 import com.tngtech.jgiven.report.AbstractReportConfig;
 import com.tngtech.jgiven.report.AbstractReportGenerator;
-import com.tngtech.jgiven.report.AbstractReportModelHandler;
-import com.tngtech.jgiven.report.AbstractReportModelHandler.ScenarioDataTable;
-import com.tngtech.jgiven.report.ReportModelHandler;
-import com.tngtech.jgiven.report.model.DataTable;
+import com.tngtech.jgiven.report.ReportBlockConverter;
 import com.tngtech.jgiven.report.model.ReportModel;
 import com.tngtech.jgiven.report.model.ReportModelFile;
-
+import com.tngtech.jgiven.report.model.ReportStatistics;
+import java.io.File;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This reporter provides the functionality for reading/writing a report in AsciiDoc format.
+ *
+ * <p> The following flags are reused from {@link AbstractReportConfig}:
+ * <ul>
+ *   <li> --format= </li>
+ *   <li> --sourceDir= / --dir= </li>
+ *   <li> --targetDir= / --todir= </li>
+ *   <li> --title= </li>
+ *   <li> --exclude-empty-scenarios=&lt;boolean&gt; </li>
+ *   <li> --help / -h </li>
+ * </ul>
+ */
 public class AsciiDocReportGenerator extends AbstractReportGenerator {
 
-    private List<String> allFiles = Lists.newArrayList();
+    private static final Logger log = LoggerFactory.getLogger(AsciiDocReportGenerator.class);
 
-    public AbstractReportConfig createReportConfig( String... args ) {
-        return new AsciiDocReportConfig( args );
-    }
+    private final List<String> featureFiles = new ArrayList<>();
+    private final List<String> failedScenarioFiles = new ArrayList<>();
+    private final List<String> pendingScenarioFiles = new ArrayList<>();
+    private final ReportBlockConverter blockConverter;
+    private File featuresDir;
+    private final Comparator<ReportModelFile> byFeatureName = Comparator.comparing(
+        modelFile -> null != modelFile.model.getName() ? modelFile.model.getName() : modelFile.model.getClassName());
 
-    public void generate() {
-        for( ReportModelFile reportModelFile : completeReportModel.getAllReportModels() ) {
-            writeReportModelToFile( reportModelFile.model, reportModelFile.file );
-        }
-        generateIndexFile();
+    public AsciiDocReportGenerator() {
```

Comment:
what is the significance of putting this into the constructor, vs just assigning it at field declaration?